bin/verifications/database/action_counts.js

```diff
@@ -0,0 +1,185 @@
+const UserModel = require('../../../models/user');
+const CommentModel = require('../../../models/comment');
+const ActionsService = require('../../../services/actions');
+const {arrayJoinBy} = require('../../../graph/loaders/util');
+const sc = require('snake-case');
+const debug = require('debug')('talk:cli:verify');
+
+const MODELS = [
+  UserModel,
+  CommentModel,
+];
+
+async function processBatch(Model, documents) {
+
+  // Get an array of all the document id's.
+  const documentIDs = documents.map(({id}) => id);
+
+  // Store all the operations on this batch in this array that we'll return
+  // later.
+  const operations = [];
+
+  // Get the action summaries for this batch.
+  const totalActionSummaries = await ActionsService
+    .getActionSummaries(documentIDs)
+    .then(arrayJoinBy(documentIDs, 'item_id'));
+
+  // Iterate over the documents.
+  for (let i = 0; i < documents.length; i++) {
+    const document = documents[i];
+    const actionSummaries = totalActionSummaries[i];
+
+    let ops = [];
+
+    for (const actionSummary of actionSummaries) {
+      if (actionSummary.group_id === null) {
+        continue;
+      }
+
+      // And we generate the group id.
+      const ACTION_TYPE = sc(actionSummary.action_type.toLowerCase());
+      const GROUP_ID = sc(actionSummary.group_id.toLowerCase());
+
+      if (GROUP_ID.length <= 0) {
+        continue;
+      }
+
+      // And we add a new batch operation if the action summary is associated
+      // with a group.
+      const ACTION_COUNT_FIELD = `${ACTION_TYPE}_${GROUP_ID}`;
+
+      // Check that the action summaries match the cached counts.
+      if (
+        !document.action_counts ||
+        !(ACTION_COUNT_FIELD in document.action_counts) ||
+        document.action_counts[ACTION_COUNT_FIELD] !== actionSummary.count
```

Comment:
This can be simplified using lodash:
`get(document, ['action_counts', ACTION_COUNT_FIELD]) !== actionSummary.count`